### DISCUSSION TITLE
Bullet point next to autocomplete suggestions

### DISCUSSION THREAD

**Discussion by stefanos82** [clangd/clangd#1826] [Q&A]
I have had a very hard time finding what are these bullet points or whatever are they called...

![image](https://github.com/clangd/clangd/assets/1777466/3c683083-fc2f-4119-97d7-5e9c47ab6834)

Does anyone know how to disable them?

**Comment by HighCommander4** [comment]
The bullet indicates that accepting the completion proposal will result in an additional text edit at the top of the file to add an `#include` for the header declaring the symbol.

If you don't want clangd to insert headers when accepting a completion proposal, the feature can be disabled by passing `--header-insertion=never` as an argument to clangd.

If you just don't want to see the bullet, that needs to be configured on the client side (i.e. it will depend on what editor you're using).

**Comment by stefanos82** [comment]
> The bullet indicates that accepting the completion proposal will result in an additional text edit at the top of the file to add an `#include` for the header declaring the symbol.
> 

That's an amazing feature I didn't know it was given to me by clangd!

> If you don't want clangd to insert headers when accepting a completion proposal, the feature can be disabled by passing `--header-insertion=never` as an argument to clangd.
> 

No, by all means I want such feature!

> If you just don't want to see the bullet, that needs to be configured on the client side (i.e. it will depend on what editor you're using).

I use Vim if it helps. Do you know anyone else who had fixed it?

I don't know how to search for it, with what keywords :confused:

**Reply by HighCommander4** [reply]
> > The bullet indicates that accepting the completion proposal will result in an additional text edit at the top of the file to add an `#include` for the header declaring the symbol.
> 
> That's an amazing feature I didn't know it was given to me by clangd!

Glad you like it. You can read more about it and other features clangd has at https://clangd.llvm.org/features and other pages of the clangd website (for management of `#include`s in particular, we have some new features in recent releases documented at https://clangd.llvm.org/guides/include-cleaner).

> > If you just don't want to see the bullet, that needs to be configured on the client side (i.e. it will depend on what editor you're using).
> 
> I use Vim if it helps. Do you know anyone else who had fixed it?
> 
> I don't know how to search for it, with what keywords 😕

I believe clients make the choice to show the bullet or not based on the presence of `additionalTextEdit`s in the [`CompletionItem`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionItem) sent by the language server.

So you might ask a question in your client's issue tracker like "how to disable showing bullets next to completion items with `additionalTextEdit`s".

In the case of vim, I understand there are a variety of forks and LSP plugins for them out there. I know [neovim](https://github.com/neovim/neovim) is a popular one, if that's what you're using then its [issue tracker](https://github.com/neovim/neovim/issues) is probably the right place to ask. (Though I'm quite unsure whether some extensions like [coc.nvim](https://github.com/neoclide/coc.nvim) might implement their own completion popups. You'll probably want to mention which plugins you're using in the issue you file.)

**Reply by stefanos82** [reply]
Thank you @HighCommander4 for your valuable feedback; I appreciate it :+1:
